onnxruntime/core/framework/session_state.cc

```diff
@@ -17,44 +17,91 @@ using namespace ::onnxruntime::common;
 
 namespace onnxruntime {
 
-const GraphViewer* SessionState::GetGraphViewer() const { return graph_viewer_.get(); }
-Status SessionState::SetGraph(const Graph& graph) {
-  graph_viewer_ = onnxruntime::make_unique<onnxruntime::GraphViewer>(graph);
-  auto& logger = Logger();
+void SessionState::SetupAllocators() {
+  for (const auto& provider : execution_providers_) {
+    for (const auto& allocator : provider->GetAllocators()) {
+      const OrtMemoryInfo& memory_info = allocator->Info();
+      if (allocators_.find(memory_info) != allocators_.end()) {
+        // EPs are ordered by priority so ignore the duplicate but log a message at ERROR level as this shouldn't
+        // happen.
+        LOGS(logger_, ERROR) << "Allocator already registered for " << allocator->Info()
```

Comment:
I'll change it to INFO. This won't stop things working. I don't think there's a good action currently given the setup where we put the EP name in the allocator resulting in per-EP rather than per-device allocators. 

---
In reply to: [443763380](https://github.com/microsoft/onnxruntime/pull/4194#discussion_r443763380) [](ancestors = 443763380)